flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -218,7 +218,10 @@
 	private final boolean enableIncrementalCheckpointing;
 
 	/** Thread number used to download from DFS when restore. */
-	private final int restoringThreadNum;
+	private final int numberOfRestoringThreads;
+
+	/** Thread number used to upload to DFS when snapshot. */
+	private final int numberOfSnapshottingThreads;
```

Comment:
Do you think it makes sense that user can configure both numbers independently? It might give more flexibility, but also makes it harder to configure or maybe a user can forget to set one of the two values. Did you experience scenarios where it made sense to have not just one number of io threads?

Reply:
@StefanRRichter In my opinion, there may be a scenario that we use multithread to restore from the savepoint but do not need to use multithread to snapshot when incremental checkpoint.

In my company, we only enabled multithread restoring. we use only RocksDBIncremental Checkpoint to snapshot and restore from the latest completed checkpoint(we do not use savepoint to restore the job and do not use HeapStateBackend in the product).